cloudslang-compiler/src/main/java/io/cloudslang/lang/compiler/modeller/SlangModellerImpl.java

```diff
@@ -57,10 +61,24 @@ public ExecutableModellingResult createModel(ParsedSlang parsedSlang) {
     private ExecutableModellingResult transformToExecutable(ParsedSlang parsedSlang, Map<String, Object> rawData) {
         String executableName = (String) rawData.get(SlangTextualKeys.EXECUTABLE_NAME_KEY);
         ExecutableModellingResult result = executableBuilder.transformToExecutable(parsedSlang, executableName, rawData);
+        validateInputNamesDifferentFromOutputNames(result);
         validateFileName(executableName, parsedSlang, result);
         return result;
     }
 
+    private void validateInputNamesDifferentFromOutputNames(ExecutableModellingResult result) {
+        List<Input> inputs = result.getExecutable().getInputs();
+        List<Output> outputs = result.getExecutable().getOutputs();
+        for (Input input : CollectionUtils.emptyIfNull(inputs)) {
+            for (Output output : CollectionUtils.emptyIfNull(outputs)) {
+                if (input.getName().equals(output.getName())) {
+                    result.getErrors().add(new RuntimeException("Inputs and outputs names should be different. " +
```

Comment:
IMHO you should also add the FQN of the executable to the error message - does it appear in a wrapper exception?